game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -749,27 +768,67 @@ public JComponent getUiComponent() {
 
       @Override
       public void save(final SaveContext context) {
-        final String username = usernameField.getText();
-        context.setValue(usernameSetting, username.isEmpty() ? null : username.toCharArray());
+        // Only save when value changed
+        if (usernameField
+            .getText()
+            .equals(usernameSetting.getValue().map(String::new).orElse(""))) {
+          return;
+        }
+        try {
+          BackgroundTaskRunner.runInBackground(
+              "Fetching Login Token...",
+              () -> {
+                final NodeBbTokenGenerator tokenGenerator = new NodeBbTokenGenerator(forumUrl);
+                final Optional<Integer> oldUserId = uidSetting.getValue();
+                final Optional<char[]> oldToken = tokenSetting.getValue();
+                if (!usernameField.getText().isEmpty()) {
+                  final TokenInfo tokenInfo =
+                      tokenGenerator.generateToken(
+                          usernameField.getText(),
+                          new String(passwordField.getPassword()),
```

Comment:
I know we had the discussion about passwords as `String` being security theater. IMO if we can keep passwords as `char[]` without too much trouble, it's not a bad thing. Said another way, if we don't have to go too far out of our way to use `char[]` for password, we probably should.